x/emissions/keeper/keeper.go

```diff
@@ -550,6 +586,24 @@ func (k *Keeper) GetLatestForecastsFromTopic(ctx context.Context, topicId TOPIC_
 	return forecasts, nil
 }
 
+// get the previous weight during rewards calculation for a topic
+func (k *Keeper) GetPreviousTopicWeight(ctx context.Context, topicId TOPIC_ID) (types.PreviousTopicWeight, error) {
+	topicWeight, err := k.previousTopicWeight.Get(ctx, topicId)
+	if errors.Is(err, collections.ErrNotFound) {
+		ret := types.PreviousTopicWeight{
```

Comment:
could just return the value?

```
if errors.Is(err, collections.ErrNotFound) {
    return types.PreviousTopicWeight{Weight: 0.0, Epoch: 0}, nil
}
```

Reply:
Don't know if the value is initialized to zero correctly, this is safer